### PR TITLE
Feature/restake topup

### PR DESCRIPTION
**Pull request by petec86**
When doing a restake the user can select to topup the amount.
Introduced a uint256 topup parameter for the restake and restakeV1 functions.

On the UI side, it should pass 0 (zero) if the user doesn't topup.

(To do: Needs Frontend UI)

### CODE REVIEW

File: contracts/Staking.sol [hunk 1]
```diff
@@ -451,10 +451,15 @@ contract Staking is IStaking, Initializable, AccessControlUpgradeable {
 
         uint256 amountOut = unstakeInternal(session, sessionId, actualEnd);
 
+        if (topup>0) {
```
Comment (Mvii7) [inline contracts/Staking.sol:454]: !=

File: contracts/Staking.sol [hunk 2]
```diff
@@ -497,6 +502,11 @@ contract Staking is IStaking, Initializable, AccessControlUpgradeable {
                 sessionStakingDays
             );
 
+        if (topup>0) {
```
Comment (Mvii7) [inline contracts/Staking.sol:505]: !=

**Comment by Mvii7** [comment]
Fix tests
Fix format

**Comment by petec86** [comment]
> Fix & add tests
If you could take over this part would be great :)

**Comment by Mvii7** [comment]
This is good to go

**Review by Mvii7 — APPROVED** [review]

**Review by Dellybro — APPROVED** [review]
Looks good to me

**Pull request merged**